WordPress/gutenberg
# Update "Edit site" admin bar link to open Site Editor at top level

Fixes [#63785](https://github.com/WordPress/gutenberg/issues/63785)

## What?
This PR updates the "Edit site" admin bar control to consistently open the Site Editor at the top level, rather than opening the specific template editing view of the current page.

## Why?
This change is necessary to improve user experience and consistency within the WordPress editor. Currently, the "Edit site" control behaves unpredictably, dropping users into various levels of the Site Editor depending on the page they're viewing. This can be disorienting and makes the control less effective as an entry point to the Site Editor.

By making this change, we align the behavior with the "Editor" control in the WP Admin > Appearance menu, providing a more predictable and consistent experience for users.

## How?
The implementation involves modifying the link destination of the "Edit site" admin bar control. Instead of linking to the specific template editing view, it will now link to the top-level Site Editor page.
```php
$wp_admin_bar->add_node(
	array(
	        'id'    => 'site-editor',
		'title' => __( 'Edit site' ),
		'href'  => admin_url( 'site-editor.php' ),
	)
);
```

## Testing Instructions
1. Ensure you're using a block-based theme.
2. Visit different pages on your site's front-end (homepage, a blog post, a static page, etc.).
3. Click the "Edit site" link in the admin bar for each of these pages.
4. Verify that regardless of which page you're on, clicking "Edit site" always takes you to the top-level Site Editor view.
5. Confirm that you're not dropped into a specific template editing view.

## Screenshots or screencast 
https://github.com/user-attachments/assets/6da0c123-8761-43bd-b67f-6644f7be99cb
